packages/astro/src/vite-plugin-astro/compile.ts

```diff
@@ -47,7 +44,6 @@ async function compile(config: AstroConfig, filename: string, source: string, vi
 	// use `sourcemap: "both"` so that sourcemap is included in the code
 	// result passed to esbuild, but also available in the catch handler.
 	const transformResult = await transform(source, {
-		as: isPage ? 'document' : 'fragment',
```

Comment:
No more `as: 'document'|'fragment'`! We no longer care, every Astro file is parsed the same.